astropy/io/fits/tests/test_hdulist.py

```diff
@@ -376,6 +376,52 @@ def test_file_like_3(self):
         info = [(0, 'PRIMARY', 1, 'PrimaryHDU', 5, (100,), 'int32', '')]
         assert fits.info(self.temp('tmpfile.fits'), output=False) == info
 
+    def test_shallow_copy(self):
+        """
+        Tests that `HDUList.__copy__()`and `HDUList.copy()` returns a
+        shallow copy (regression test for ticket:7211).
+        """
+
+        n = np.arange(10.0)
+        primary_hdu = fits.PrimaryHDU(n)
+        hdu = fits.ImageHDU(n)
+        hdul = fits.HDUList([primary_hdu, hdu])
+
+        shallow_copy = hdul.copy()
+        magic_shallow_copy = hdul.__copy__()
```

Comment:
You should test with `copy.copy` (and `copy.deepcopy` below) instead of using the magic methods.

Reply:
I am not sure what you mean by "test with `copy.copy`". Why do we need to use `copy` module here instead of testing the methods we added?